src/dotnet/commands/dotnet-new3/HelpFormatter.cs

```diff
@@ -0,0 +1,228 @@
+            }
+        }
+
+        private class TextWrapper
+        {
+            private readonly IReadOnlyList<string> _lines;
+
+            public TextWrapper(string text, int maxWidth, bool alwaysMax)
+            {
+                List<string> lines = new List<string>();
+                int position = 0;
+                int realMaxWidth = alwaysMax ? maxWidth : 0;
+
+                while (position < text.Length)
+                {
+                    int newline = text.IndexOf(Environment.NewLine, position, StringComparison.Ordinal);
+
+                    if (newline > -1)
+                    {
+                        if (newline - position <= maxWidth)
+                        {
+                            lines.Add(text.Substring(position, newline - position).TrimEnd());
+                            position = newline + Environment.NewLine.Length;
+                        }
+                        else
+                        {
+                            GetLineText(text, lines, maxWidth, newline, ref position);
+                        }
+                    }
+                    else
+                    {
+                        GetLineText(text, lines, maxWidth, text.Length - 1, ref position);
+                    }
+
+                    realMaxWidth = Math.Max(realMaxWidth, lines[lines.Count - 1].Length);
+                }
+
+                _lines = lines;
+                MaxWidth = realMaxWidth;
+            }
+
+            public int LineCount => _lines.Count;
+
+            public int MaxWidth { get; }
+
+            public string this[int index, int padTo = 0]
+            {
+                get { return (_lines.Count > index ? _lines[index] : string.Empty).PadRight(MaxWidth).PadRight(padTo > MaxWidth ? padTo : MaxWidth); }
+            }
+
+            private static void GetLineText(string text, List<string> lines, int maxLength, int end, ref int position)
+            {
+                if (text.Length - position < maxLength)
+                {
+                    lines.Add(text.Substring(position));
+                    position = text.Length;
+                    return;
+                }
+
+                int lastBreak = text.LastIndexOfAny(new[] { ' ', '-' }, end, end - position);
+                while (lastBreak > 0 && lastBreak - position > maxLength)
+                {
+                    --lastBreak;
+                    lastBreak = text.LastIndexOfAny(new[] { ' ', '-' }, lastBreak, lastBreak - position);
+                }
+
+                if (lastBreak > 0)
+                {
+                    lines.Add(text.Substring(position, lastBreak - position + 1).TrimEnd());
+                    position = lastBreak + 1;
+                }
+                else
+                {
+                    int properMax = Math.Min(maxLength - 1, text.Length - position);
+                    lines.Add(text.Substring(position, properMax) + '-');
+                    position += maxLength - 1;
```

Comment:
+=properMax? seems like otherwise position might be messed up somewhere higher in the stack